packages/commons/core-utils/src/mergeWithOverrides.ts

```diff
@@ -1,7 +1,20 @@
 import type { Dictionary, NumericDictionary, PartialObject, PropertyName, ValueKeyIteratee } from "lodash";
 import { isNull, isPlainObject, mergeWith, omitBy } from "lodash-es";
 
-export function mergeWithOverrides<T extends object>({ data, overrides }: { data: T; overrides: object }): T {
+export type AncestorOmissionCriteria = {
```

Comment:
nit: rename to `NullOmissionOptions` ?

Reply:
OmitDeepBy should accept `AncestorOmissionOptions` but `mergeWithOverrides` should just expose a `NullOmissionOptions`